src/client/datascience/editor-integration/codeLensFactory.ts

```diff
@@ -272,6 +272,11 @@ export class CodeLensFactory implements ICodeLensFactory, IInteractiveWindowList
         args.notebook.onDisposed(() => {
             this.notebookData.delete(key);
         });
+        if (args.notebook.onDidFinishExecuting !== undefined) {
+            args.notebook.onDidFinishExecuting((cell: ICell) => {
+                this.updateExecutionCounts(args.identity, cell);
+            });
+        }
```

Comment:
Unfortunately codeLensFactory.ts uses `ICell` all over the place. We should rewrite the whole thing to just work with `NotebookCell`s and change how we execute cells for the interactive window to execute `NotebookCell`s rather than using the old `executeObservable` codepath.